File: 준영/비밀지도.py
def to_bin(n, x):
    result = ''
    N = n
    n -= 1
    for _ in range(N):
        result += str(x // (2**n))
        x %= (2**n)
        n -= 1
    return result

def solution(n, arr1, arr2):
    answer = []
    for i in range(n):
        temp = ""
        x = to_bin(n, arr1[i])
        y = to_bin(n, arr2[i])
        for j in range(n):
            if x[j] == '1' or y[j] == '1':
                temp += "#"
            else:
                temp += " "
                
        answer.append(temp)
            
    return answer

solution(5, [9, 20, 28, 18, 11], [30, 1, 21, 17, 28])

File: 준영/LV1/비밀지도.js
function to_bin(n, x) {
  var result = []
  for (let i = 0; i < n; i++) {
    if (parseInt(x / (2**(n-1-i))) === 1) {
      result.push("1")
    } else {
      result.push("0")
    }
    x %= (2**(n-1-i))
  }
  return result
}

function solution(n, arr1, arr2) {
  var answer = [];
  for (let i = 0; i < n; i++) {
    var temp = ""
    var x = to_bin(n, arr1[i])
    var y = to_bin(n, arr2[i])
    for (let j = 0; j < n; j++) {
      if (x[j] === '1' || y[j] === '1') {
        temp += "#"
      } else {
        temp += " "
      } 
    }
    answer.push(temp)
  }
  return answer;
}